client/extensions/woocommerce/state/sites/settings/email/test/actions.js

```diff
@@ -138,4 +150,195 @@ describe( 'actions', () => {
+			submitMailChimpStoreInfo( siteId, storeInfo )( dispatch, getState );
+			expect( dispatch ).to.have.been.calledWith( {
+				type: WOOCOMMERCE_MAILCHIMP_STORE_INFO_SUBMIT,
+				siteId,
+			} );
+		} );
+
+		test( 'should dispatch a success action with settings information when request completes', () => {
+			const getState = () => ( {} );
+			const dispatch = spy();
+			const storeInfo = {
+				store_name: 'mystore',
+				store_street: 'street',
+				store_city: 'Valhalla',
+			};
+			const response = submitMailChimpStoreInfo( siteId, storeInfo )( dispatch, getState );
+
+			return response.then( () => {
+				expect( dispatch ).to.have.been.calledWith( {
+					type: WOOCOMMERCE_MAILCHIMP_STORE_INFO_SUBMIT_SUCCESS,
+					siteId,
+					settings: {
+						mailchimp_api_key: '12345testing',
+						mailchimp_debugging: false,
+						mailchimp_account_info_id: '64c533c4',
+						mailchimp_account_info_username: 'ski',
+						mailchimp_active_tab: 'campaign_defaults',
+						store_name: 'mystore',
+						store_street: 'street',
+						store_city: 'Valhalla',
+					},
+				} );
+			} );
+		} );
+	} );
+
+	describe( '#submitMailChimpCampaignDefaults()', () => {
+		const siteId = '123';
+
+		useNock( nock => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
+				.post( '/rest/v1.1/jetpack-blogs/123/rest-api/' )
+				.query( { path: '/wc/v3/mailchimp/store_info&_method=put', json: true } )
+				.reply( 200, {
+					data: {
+						mailchimp_api_key: '12345testing',
+						mailchimp_debugging: false,
+						mailchimp_account_info_id: '64c533c4',
+						mailchimp_account_info_username: 'ski',
+						mailchimp_active_tab: 'campaign_defaults',
+						store_name: 'mystore',
+						store_street: 'street',
+						store_city: 'Valhalla',
+						campaign_from_name: 'woo',
+						campaign_from_email: 'mystore@woo.woo',
+					},
+				} );
+		} );
+
+		test( 'should dispatch an action', () => {
+			const getState = () => ( {} );
+			const dispatch = spy();
+			const campaignDefaults = {
+				campaign_from_name: 'woo',
+				campaign_from_email: 'mystore@woo.woo',
+			};
+			submitMailChimpCampaignDefaults( siteId, campaignDefaults )( dispatch, getState );
+			expect( dispatch ).to.have.been.calledWith( {
+				type: WOOCOMMERCE_MAILCHIMP_CAMPAIGN_DEFAULTS_SUBMIT,
+				siteId,
+			} );
+		} );
+
+		test( 'should dispatch a success action with settings information when request completes', () => {
+			const getState = () => ( {} );
+			const dispatch = spy();
+			const campaignDefaults = {
+				campaign_from_name: 'woo',
+				campaign_from_email: 'mystore@woo.woo',
+			};
+			const response = submitMailChimpCampaignDefaults( siteId, campaignDefaults )(
+				dispatch,
+				getState
+			);
+
+			return response.then( () => {
+				expect( dispatch ).to.have.been.calledWith( {
+					type: WOOCOMMERCE_MAILCHIMP_CAMPAIGN_DEFAULTS_SUBMIT_SUCCESS,
+					siteId,
+					settings: {
+						mailchimp_api_key: '12345testing',
```

Comment:
Same optional comment as above regarding a constant for nock and action creator instead of repeating